fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -913,10 +966,11 @@ struct EncryptedRangeFileWriter : public IRangeFileWriter {
 private:
 	Standalone<StringRef> buffer;
 	uint8_t* wPtr;
-	BlobCipherEncryptHeader* encryptHeader;
+	uint8_t* encryptHeader;
```

Comment:
What `arena` controls the lifecycle of the allocated buffer? `BlobCipherEncryptHeader` had all fixed sized `structs` so just storing as a pointer would have worked, however, with `BlobCipherEncyrptHeaderRef` it might not be possible (due to StringRef component); would it be better and/or safer to store `StringRef` instead?

Reply:
I make the arena for the `BlobCipherEncyrptHeaderRef`  dependent on the one which we pass to the FileWriter. It's something similar to what you do for blob

Reply:
And i anyways have to copy the header from Standalone you return to it's correct location in the backup block

Reply:
But storing `naked pointer` seems risky; why don't we just make it `Standalone<StringRef>` ?

Reply:
Also, where is associated `arena`? I couldn't find it as one of the class data members...

Reply:
> It's something similar to what you do for blob
For BlobGranule, all allocations are done on client supplied `arena`, plus, the in-memory classes stores `StringRef` associated with the same `arena` object. `BlobCipherEncryptHeaderRef` when serialized to stringRef using `arena.dependOn()` to attach Standalone<StringRef> lifecycle gets associated with client supplied arena

Reply:
Because i need to leave space for the header in the backup block, then perform the backup then afterwards i need to know where to enter the header once i've done the encryption

Reply:
Its not necessarily a naked pointer its just a pointer to where in the backup block the header should go

Reply:
Am not sure why it can't be a `StringRef`?

Reply:
Spoke offline will wrap in a StringRef to make it clearer